### PR TITLE
add missing page_info / cursors for Association Proxy logic 

### PR DESCRIPTION
**Pull request by Ckk3**
**Note:** This PR **supersedes** #232 because the original contributor did not allow edits from maintainers, and we needed to add additional tests to proceed with the merge. All credits remain to the original author for the core fix.

Original description:

## Description

`association_proxy_resolver_for` has a few code blocks that were not updated for the relay changes. Association proxy mapping would fail due do constructors missing either `page_info` or `cursor` args. 

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Documentation

## Issues Fixed or Closed by This PR

* N/a

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

I haven't added tests as it looks like this code is originally untested.

## Summary by Sourcery

Bug Fixes:
- Fix association proxy mapping failures due to missing `page_info` or `cursor` arguments in constructors.

## Summary by Sourcery

Bug Fixes:
- Ensure association proxy resolvers return valid relay connections, including `page_info` and edge `cursor` details, even for empty results.

## Summary by Sourcery

Fix association proxy resolvers to always return valid relay connections with correct page_info and cursor fields, including for empty results, and add comprehensive tests for association proxy logic.

Bug Fixes:
- Fix association proxy mapping failures by ensuring relay connections include page_info and cursor fields, even for empty results.

Enhancements:
- Improve error messages for unsupported association proxy targets and related exceptions.

Documentation:
- Add release notes describing the patch and its impact.

Tests:
- Add extensive tests for association proxy behavior, including edge cases and nullable relationships.

### CODE REVIEW

**Comment by Ckk3** [comment]
@sourcery-ai review

**Comment by Ckk3** [comment]
@sourcery-ai summary

File: src/strawberry_sqlalchemy_mapper/mapper.py [hunk 1]
```diff
@@ -97,13 +95,14 @@
...
+    from sqlalchemy.orm.state import InstanceState
     from sqlalchemy.sql.expression import ColumnElement
+    from strawberry.types.field import StrawberryField
 
 BaseModelType = TypeVar("BaseModelType")
 
 SkipTypeSentinelT = NewType("SkipTypeSentinelT", object)
-SkipTypeSentinel = cast(SkipTypeSentinelT, sentinel.create("SkipTypeSentinel"))
-
+SkipTypeSentinel = cast("SkipTypeSentinelT", sentinel.create("SkipTypeSentinel"))
```
[the remark this answers is not in the document]
Comment (Ckk3) [reply src/strawberry_sqlalchemy_mapper/mapper.py:105]: We're already using the real SkipTypeSentinelT type inside a TYPE_CHECKING block to preserve static typing while avoiding runtime import issues. The string in cast() is there to keep Ruff happy with TC002.

File: tests/test_association_proxy.py [hunk 2]
```diff
@@ -0,0 +1,659 @@
...
+              }
+            }
+          }
+        }
+      }
+    }
+    """
+
+
+async def test_query_with_association_proxy_schema(
```
[the remark this answers is not in the document]
Comment (Ckk3) [reply tests/test_association_proxy.py:321]: Added thank you!

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

Ensure association proxy resolvers return valid relay connections, including `page_info` and edge `cursor` details, even for empty results.

**Comment by Ckk3** [comment]
Ready for review, @patrick91 @bellini666 @fruitymedley !

**Comment by Ckk3** [comment]
@sourcery-ai review

File: src/strawberry_sqlalchemy_mapper/mapper.py [hunk 3]
```diff
@@ -789,7 +802,8 @@ def convert(type_: Any) -> Any:
             # ignore inherited `is_type_of`
             if "is_type_of" not in type_.__dict__:
                 type_.is_type_of = (
-                    lambda obj, info: type(obj) == model or type(obj) == type_
+                    lambda obj, info: type(obj) == model  # noqa: E721
+                    or type(obj) == type_  # noqa: E721
                 )
```
[the remark this answers is not in the document]
Comment (Ckk3) [reply src/strawberry_sqlalchemy_mapper/mapper.py:807]: I prefer maintain the older one.

**Review by patrick91 — APPROVED** [review]

**Comment by Ckk3** [comment]
Thanks @tylernisonoff 😊
